### PR TITLE
Implement old data reporting

### LINKED ISSUE #11435: Add option for DBSync to report the old contents of a registry
Hello team!

While developing #9103, we would need an option for DBSync to report the old contents of a registry when it's modified.

## Use case

When FIM sends a modification event, it sends the old attributes of the file and analysisd uses this information to create the alert. In case that DBSync detects a change in a registry, only reports the difference between the two snapshots.

### PR DESCRIPTION
**Pull request by RamiroRD**
(Closes #11435)
## Description

This PR applies changes to DbSync::syncRow() to optionally return the previous values of rows that were updated. Now DbSync::syncRow checks for a "return_old_data" boolean field, which determines whether we should return the diff or not when syncing a row.

- Compilation without warnings in every supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [ ] Source installation
- [ ] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

- Decoder/Rule tests
  - [ ] Added unit testing files ".ini"
  - [ ] runtests.py executed without errors

### CODE REVIEW

File: src/shared_modules/dbsync/include/dbsync.h [hunk 1]
```diff
@@ -161,7 +161,7 @@ EXPORTED int dbsync_sync_row(const DBSYNC_HANDLE handle,
  * @brief Select data, based in \p json_data_input data, from the database table.
  *
  * @param handle          Handle assigned as part of the \ref dbsync_create method().
- * @param js_data_input   JSON with table name, fields and filters to apply in the query.
+ * @param js_data_input   JSON with table name, fields, filters and options to apply in the query.
```
Comment (RamiroRD) [inline src/shared_modules/dbsync/include/dbsync.h:164]: Where are these filters documented? I should modify whatever documentation we have to reflect these changes.

File: src/shared_modules/dbsync/src/dbsync_implementation.cpp [hunk 2]
```diff
@@ -49,14 +49,28 @@ void DBSyncImplementation::insertBulkData(const DBSYNC_HANDLE   handle,
     ctx->m_dbEngine->bulkInsert(json.at("table"), json.at("data"));
 }
 
+static bool shouldReturnOldData(const nlohmann::json& data)
+{
+    auto it { data.find("return_old_data") };
+
+    if (it != data.end())
+    {
+        return *it;
```
Comment (HanesSciarrone) [inline src/shared_modules/dbsync/src/dbsync_implementation.cpp:58]: Hi @RamiroRD 

TFor RVO C++ is better only using one return.

Best regards
Hanes

**Review by HanesSciarrone — CHANGES_REQUESTED** [review]

File: src/shared_modules/dbsync/src/dbsync_implementation.cpp (hunk 2 above)
Comment (RamiroRD) [reply src/shared_modules/dbsync/src/dbsync_implementation.cpp:58]: Right, I missed this one. Thanks

**Review by HanesSciarrone — APPROVED** [review]

File: src/shared_modules/dbsync/src/dbsync_implementation.cpp [hunk 3]
```diff
@@ -49,14 +49,29 @@ void DBSyncImplementation::insertBulkData(const DBSYNC_HANDLE   handle,
...
+    bool result {false};
+
+    if (it != data.end())
+    {
+        result = *it;
+    }
+
+    return result;
+}
+
```
[the remark this answers is not in the document]
Comment (jnasselle) [reply src/shared_modules/dbsync/src/dbsync_implementation.cpp:64]: Nevermind. We also need the value for the option

**Review by jnasselle — APPROVED** [review]

**Review by antoniomanuelfr — APPROVED** [review]
I have been testing the feature in a developing branch and is working perfectly

**Comment by RamiroRD** [comment]
Moved to https://github.com/wazuh/wazuh/pull/12094/files

**Pull request closed without merging**